src/ILCompiler.Compiler/src/IL/Stubs/Bootstrap/BootstrapMethod.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Internal.TypeSystem;
+using Internal.IL;
+using Internal.IL.Stubs;
+
+namespace ILCompiler.Compiler.IL.Stubs
+{
+    /// <summary>
+    /// Bootstrap method base for IL methods that run during bootstrapping the runtime.
+    /// </summary>
+    public abstract class BootstrapMethod : ILStubMethod
+    {
```

Comment:
I think this is unnecessary abstraction, and it does not have much to do with bootstrap. I would fold it with BootstrapMainMethod.

If you would like to reduce amount of boilerplate code here - you can add this to `class ILStubMethod`:

```
public override TypeSystemContext Context  
{  
    get  
    {  
        return OwningType.Context;  
    }
}
```